### ISSUE TITLE
no-carrier sometimes not picked up

### ISSUE THREAD

**Issue by nberlee** [bpetlert/networkd-broker#1]
I started networkd-broker with -vvv to get the linkEvents and `sudo busctl monitor org.freedesktop.network1`  to catch all events on the dbus.

Sometimes, when I unplug by ethernet cable or disable Wifi (hardware button) networkd-broker is not picking up the last `no-carrier` or `off` status

DBus messages when `no-carrier` is ignored:
```
‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=349
  Sender=:1.4489  Path=/org/freedesktop/network1/link/_32  Interface=org.freedesktop.DBus.Properties  Member=PropertiesChanged
  UniqueName=:1.4489
  MESSAGE "sa{sv}as" {
          STRING "org.freedesktop.network1.Link";
          ARRAY "{sv}" {
                  DICT_ENTRY "sv" {
                          STRING "CarrierState";
                          VARIANT "s" {
                                  STRING "no-carrier";
                          };
                  };
                  DICT_ENTRY "sv" {
                          STRING "OperationalState";
                          VARIANT "s" {
                                  STRING "no-carrier";
                          };
                  };
          };
          ARRAY "s" {
          };
  };

‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=350
  Sender=:1.4489  Path=/org/freedesktop/network1/link/_32  Interface=org.freedesktop.DBus.Properties  Member=PropertiesChanged
  UniqueName=:1.4489
  MESSAGE "sa{sv}as" {
          STRING "org.freedesktop.network1.Link";
          ARRAY "{sv}" {
                  DICT_ENTRY "sv" {
                          STRING "AddressState";
                          VARIANT "s" {
                                  STRING "degraded";
                          };
                  };
          };
          ARRAY "s" {
          };
  };
```

DBus messages when `no-carrier`  get picked up by networkd-broker:
```
‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=499
  Sender=:1.4489  Path=/org/freedesktop/network1/link/_32  Interface=org.freedesktop.DBus.Properties  Member=PropertiesChanged
  UniqueName=:1.4489
  MESSAGE "sa{sv}as" {
          STRING "org.freedesktop.network1.Link";
          ARRAY "{sv}" {
                  DICT_ENTRY "sv" {
                          STRING "CarrierState";
                          VARIANT "s" {
                                  STRING "no-carrier";
                          };
                  };
                  DICT_ENTRY "sv" {
                          STRING "OperationalState";
                          VARIANT "s" {
                                  STRING "no-carrier";
                          };
                  };
          };
          ARRAY "s" {
          };
  };

‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=500
  Sender=:1.4489  Path=/org/freedesktop/network1/link/_32  Interface=org.freedesktop.DBus.Properties  Member=PropertiesChanged
  UniqueName=:1.4489
  MESSAGE "sa{sv}as" {
          STRING "org.freedesktop.network1.Link";
          ARRAY "{sv}" {
                  DICT_ENTRY "sv" {
                          STRING "AddressState";
                          VARIANT "s" {
                                  STRING "degraded";
                          };
                  };
          };
          ARRAY "s" {
          };
  };

‣ Type=method_call  Endian=l  Flags=0  Version=1 Cookie=2
  Sender=:1.4628  Destination=org.freedesktop.network1  Path=/org/freedesktop/network1/link/_32  Interface=org.freedesktop.DBus.Properties  Member=Get
  UniqueName=:1.4628
  MESSAGE "ss" {
          STRING "org.freedesktop.network1.Link";
          STRING "BitRates";
  };

‣ Type=method_return  Endian=l  Flags=1  Version=1 Cookie=501  ReplyCookie=2
  Sender=:1.4489  Destination=:1.4628
  UniqueName=:1.4489
  MESSAGE "v" {
          VARIANT "(tt)" {
                  STRUCT "tt" {
                          UINT64 18446744073709551615;
                          UINT64 18446744073709551615;
                  };
          };
  };

‣ Type=method_call  Endian=l  Flags=0  Version=1 Cookie=3
  Sender=:1.4628  Destination=org.freedesktop.network1  Path=/org/freedesktop/network1/link/_32  Interface=org.freedesktop.DBus.Properties  Member=Get
  UniqueName=:1.4628
  MESSAGE "ss" {
          STRING "org.freedesktop.network1.DHCPServer";
          STRING "Leases";
  };

‣ Type=error  Endian=l  Flags=1  Version=1 Cookie=502  ReplyCookie=3
  Sender=:1.4489  Destination=:1.4628
  ErrorName=org.freedesktop.DBus.Error.UnknownProperty  ErrorMessage="Unknown interface org.freedesktop.network1.DHCPServer or property Leases."
  UniqueName=:1.4489
  MESSAGE "s" {
          STRING "Unknown interface org.freedesktop.network1.DHCPServer or property Leases.";
  };
```
This is picked up by networkd_broker because I see:
```
DEBUG networkd_broker::broker > LinkEvent { path: Path("/org/freedesktop/network1/link/_32"), state_type: OperationalState, state: NoCarrier }
INFO  networkd_broker::broker > Respond to 'no-carrier' event of 'enp0s31f6'
```

`networkd-notify` is notifying everytime, even if networkd-broker choose to ignore the dbus message

**Comment by bpetlert** [comment]
Thank you for reporting this issue. I am investigating it.

**Comment by bpetlert** [comment]
This issue should be fixed in version 0.4.x release, do let me know if you are still able to reproduce this issue.
